packages/excalidraw/components/App.tsx

```diff
@@ -9102,25 +9105,60 @@ class App extends React.Component<AppProps, AppState> {
           this.state,
         );
 
-        if (!pointerDownState.drag.hasOccurred && newElement && !multiElement) {
-          this.scene.mutateElement(
-            newElement,
-            {
-              points: [
-                ...newElement.points,
-                pointFrom<LocalPoint>(
-                  pointerCoords.x - newElement.x,
-                  pointerCoords.y - newElement.y,
-                ),
-              ],
-            },
-            { informMutation: false, isDragging: false },
-          );
+        const dragDistance =
+          pointDistance(
+            pointFrom(pointerCoords.x, pointerCoords.y),
+            pointFrom(pointerDownState.origin.x, pointerDownState.origin.y),
+          ) * this.state.zoom.value;
 
-          this.setState({
-            multiElement: newElement,
-            newElement,
-          });
+        if (
+          (!pointerDownState.drag.hasOccurred ||
+            dragDistance < MINIMUM_ARROW_SIZE) &&
+          newElement &&
+          !multiElement
+        ) {
+          if (this.device.isTouchScreen) {
+            const FIXED_DELTA_X = Math.min(
+              (this.state.width * 0.7) / this.state.zoom.value,
+              100,
+            );
+
+            this.scene.mutateElement(
+              newElement,
+              {
+                x: newElement.x - FIXED_DELTA_X / 2,
+                points: [
+                  pointFrom<LocalPoint>(0, 0),
+                  pointFrom<LocalPoint>(FIXED_DELTA_X, 0),
+                ],
+              },
+              { informMutation: false, isDragging: false },
+            );
+
+            this.actionManager.executeAction(actionFinalize);
+            this.setState({
+              editingLinearElement: new LinearElementEditor(
+                newElement,
+                this.scene.getNonDeletedElementsMap(),
+              ),
+            });
```

Comment:
this one currently has bad UX with pen input because it's pretty much impossible to exit the line editor by pointerdown with a pen. Not sure if it's because of the pen immediately box-selecting due to being oversensitive, or if we really prevent pen events to exit line editing. So we should fix it (e.g. add bigger threshold for box-selecting to not exit the editor while in line editor if that's the root cause).

NOTE that in prod it's still hard to exit with pen, but somewhat doable. In this PR it seems impossible, but I can't see which code change in this PR might be causing this...

Reply:
Yeah, the current UX for using a pen in line editing mode, from what I can tell, is to allow the users to
- use the pen to box select points
- clear the selected points on pen down
- move the selected points
And it seems that we can only exit the line editing mode with our fingers. 

Not noticing any differences between prod and this PR. But will investigate a bit more.